e2e/libs/node/operations/operations.py

```diff
@@ -0,0 +1,166 @@
+import time
+import logging
+import subprocess
+
+from functools import partial
+from multiprocessing import Pool
+from strategy import CloudProvider
+from node.operations.aws import EC2
+from node.operations.local_cluster import LocalCluster
+from utility import globalVars, Utility
+from volume.rest_volume import RestVolume
+
+class Operations:
+
+    _instance = None
+
+    def __init__(self, cloud_provider):
+        if cloud_provider == CloudProvider.AWS.value:
+            logging.info("cloude provider: AWS")
+            self.__class__._instance  = EC2()
+        elif cloud_provider == CloudProvider.LOCAL_CLUSTER.value:
+            logging.info("cloude provider: Local")
+            self.__class__._instance = LocalCluster()
+        else:
+            Exception(f"could not recognize the cloud provider: {cloud_provider}")
+
+    @classmethod
+    def cleanup(cls):
+        # Turn the power off node back
+        cls._instance.power_on_node_instance()
+
+    @classmethod
+    def power_off_node(cls, node_name):
+        cls._instance.power_off_node_instance(node_name)
+
+    @classmethod
+    def power_on_node(cls, node_name):
+        cls._instance.power_on_node_instance(node_name)
+
+    @classmethod
+    def reboot_node(cls, node_name):
+        cls._instance.reboot_node_instance(node_name)
+
+    @classmethod
+    def restart_kubelet(cls, node_name, interval_time):
+        if node_name == "":
+            raise Exception("failed with empty parameter value: node_name")
+
+        logging.info(f"restarting kubelet on node instances: {node_name}")
+        node_instances = cls._instance.get_node_instance(node_name)
+
+        k8s_distro = globalVars.variables["K8S_DISTRO"]
+        if k8s_distro == 'rke2':
+            stopCmd = 'sudo systemctl stop rke2-server.service'
+            startCmd = 'sudo systemctl start rke2-server.service'
+            restart = 'sudo systemctl restart rke2-server.service'
+        elif k8s_distro == 'rke1':
+            stopCmd = 'sudo docker stop kubelet'
+            startCmd = 'sudo docker run kubelet'
+            restart = 'sudo docker restart kubelet'
+        elif k8s_distro == 'k3s':
+            stopCmd = 'systemctl stop k3s-agent.service'
+            startCmd = 'systemctl start k3s-agent.service'
+            restart = 'systemctl restart k3s-agent.service'
+        else:
+            raise Exception(f'Unsupported K8S distros: {k8s_distro}')
+
+        if int(interval_time) == 0:
+            for instance in node_instances:
+                ip_address = instance.public_ip_address
+                logging.debug(f'ip_address:{ip_address}, cmd:{restart}')
```

Comment:
Do we still need this? I think you already has simulate log in the Utility.ssh_and_exec_cmd.